kotlin-runtime/ftl-plugin/src/main/kotlin/xyz/block/ftl/gradle/SchemaGenerator.kt

```diff
@@ -0,0 +1,162 @@
+package xyz.block.ftl.gradle
+
+import com.squareup.kotlinpoet.*
+import com.squareup.wire.GrpcClient
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.SendChannel
+import okhttp3.OkHttpClient
+import okhttp3.Protocol
+import org.gradle.configurationcache.extensions.capitalized
+import xyz.block.ftl.v1.ControllerServiceClient
+import xyz.block.ftl.v1.PullSchemaRequest
+import xyz.block.ftl.v1.PullSchemaResponse
+import xyz.block.ftl.v1.schema.Type
+import java.time.Duration
+
+class SchemaGenerator(ftlEndpoint: String) {
+  private var grpcClient: GrpcClient
+  private lateinit var sendSchemaChannel: SendChannel<PullSchemaRequest>
+  private lateinit var receiveSchemaChannel: ReceiveChannel<PullSchemaResponse>
+
+  init {
+    grpcClient = GrpcClient.Builder()
+      .client(
+        OkHttpClient.Builder()
+          .readTimeout(Duration.ofSeconds(10))
+          .writeTimeout(Duration.ofSeconds(10))
+          .callTimeout(Duration.ofSeconds(10))
+          .protocols(listOf(Protocol.H2_PRIOR_KNOWLEDGE))
+          .build()
+      )
+      .baseUrl(ftlEndpoint)
+      .build()
+  }
+
+  fun generate() {
+    runBlocking {
+      launch {
+        grpcClient.create(ControllerServiceClient::class)
```

Comment:
I'd probably split this into two classes: one that pulls the schema from the server and an `internal` one that just accepts a proto-encoded schema and writes Kotlin code.